core/new-gui/src/app/dashboard/component/feature-container/saved-workflow-section/saved-workflow-section.component.ts

```diff
@@ -997,4 +969,128 @@ export class SavedWorkflowSectionComponent implements OnInit, OnChanges {
+   */
+  private handleFileUploads(file: Blob, name: string) {
+    let reader = new FileReader();
+    reader.readAsText(file);
+    reader.onload = () => {
+      try {
+        const result = reader.result;
+        if (typeof result !== "string") {
+          throw new Error("Incorrect format: file is not a string");
+        }
+        const workflowContent = JSON.parse(result) as WorkflowContent;
+        const fileExtensionIndex = name.lastIndexOf(".");
+        let workflowName: string;
+        if (fileExtensionIndex === -1) {
+          workflowName = name;
+        } else {
+          workflowName = name.substring(0, fileExtensionIndex);
+        }
+        if (workflowName.trim() === "") {
+          workflowName = DEFAULT_WORKFLOW_NAME;
+        }
+        this.workflowPersistService
+          .createWorkflow(workflowContent, workflowName)
+          .pipe(untilDestroyed(this))
+          .subscribe({
+            next: uploadedWorkflow => {
+              this.dashboardWorkflowEntries = [...this.dashboardWorkflowEntries, uploadedWorkflow];
+            },
+            error: (err: unknown) => alert(err),
+          });
+      } catch (error) {
+        this.notificationService.error(
+          "An error occurred when importing the workflow. Please import a workflow json file."
+        );
+        console.error(error);
+      }
+    };
+  }
+
+  /**
+   * Download selected workflow as zip file
+   */
+  public onClickOpenDownloadZip() {
+    let dateTime = new Date();
+    let filename = "workflowExports" + dateTime.toISOString() + ".zip";
+    this.zip.generateAsync({ type: "blob" }).then(function (content) {
+      saveAs(content, filename);
+    });
+  }
+
+  /**
+   * Adding the workflow as pending download zip file
+   */
+  public onClickAddToDownload(dashboardWorkflowEntry: DashboardWorkflowEntry, event: Event) {
+    console.log(dashboardWorkflowEntry.workflow.wid);
+    console.log((<HTMLInputElement>event.target).checked);
+    if ((<HTMLInputElement>event.target).checked) {
+      if (dashboardWorkflowEntry.workflow.wid) {
+        if (!this.downloadListWorkflow.includes(dashboardWorkflowEntry.workflow.wid)) {
+          this.downloadListWorkflow.push(dashboardWorkflowEntry.workflow.wid);
+          this.notificationService.success(
+            "Successfully add workflow " + dashboardWorkflowEntry.workflow.wid + " to download list."
+          );
+        }
+        this.workflowPersistService
+          .retrieveWorkflow(dashboardWorkflowEntry.workflow.wid)
+          .pipe(untilDestroyed(this))
+          .subscribe(data => {
+            const workflowCopy: Workflow = {
+              ...data,
+              wid: undefined,
+              creationTime: undefined,
+              lastModifiedTime: undefined,
+            };
+            const workflowJson = JSON.stringify(workflowCopy.content);
+            const fileName = dashboardWorkflowEntry.workflow.name + "-" + dashboardWorkflowEntry.workflow.wid + ".json";
```

Comment:
I remember from our group discussion we decided the filename should not include wid? Instead whenever there is a duplicate name we add a count after.

Reply:
Done.

Reply:
The fix I propose only uses regex and thus did not work very well when similar workflow name exist, for example, "Workflow1" and "Workflow12", when checked together, the latest checked will have a count after. (Workflow1, Workflow12-1) This is the limitation when use JSzip, the lib did not provide a direct way to check if the file already exists, I could only check the regex and it would return the number of files matches with the regex. Should I still use this implementation or revert back to the wid method which makes more sense?

Reply:
Maybe you can check each existing file names when adding a new file, and make necessary changes. I do not think regex is needed, just whole string equality comparison, case sensitive, to avoid having two files with the exact same name.
Another thought: if you just leave the duplicate name as it is, what would the library do? Maybe it already automatically handles that for you?

Reply:
Changed method.